tf2_geometry_msgs/scripts/test.py

```diff
@@ -135,6 +135,42 @@ def test_transform(self):
         self.assertEqual(out.vector.y, 0)
         self.assertEqual(out.vector.z, 0)
 
+        # Testing for pose and covariance transform
+        t = TransformStamped()
+        t.transform.translation.x = 1.0
+        t.transform.translation.y = 2.0
+        t.transform.translation.z = 3.0
+        t.transform.rotation = Quaternion(w=0.0, x=1.0, y=0.0, z=0.0)
+
+        v = PoseWithCovarianceStamped()
+        v.header.stamp = rclpy.time.Time(seconds=2.0).to_msg()
+        v.header.frame_id = 'a'
+        v.pose.covariance = (
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0
+        )
+        v.pose.pose.position.x = 1.0
+        v.pose.pose.position.y = 2.0
+        v.pose.pose.position.z = 3.0
+        v.pose.pose.orientation = Quaternion(w=0.0, x=1.0, y=0.0, z=0.0)
+
+        out = tf2_geometry_msgs.do_transform_pose_with_covariance_stamped(v, t)
+        expected_covariance = (
+          1.0, 0.0, 0.0, 4.0, 0.0, 0.0,
+          0.0, 2.0, 0.0, 0.0, 5.0, 0.0,
+          0.0, 0.0, 3.0, 0.0, 0.0, 6.0,
+          1.0, 0.0, 0.0, 4.0, 0.0, 0.0,
+          0.0, 2.0, 0.0, 0.0, 5.0, 0.0,
+          0.0, 0.0, 3.0, 0.0, 0.0, 6.0
+        )
```

Comment:
Can you double check the math here? I did some matrix multiplications by hand, and I think this should instead read:
```suggestion
        expected_covariance = (
          1.0, -2.0, -3.0, 4.0, -5.0, -6.0,
          -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
          -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
          1.0, -2.0, -3.0, 4.0, -5.0, -6.0,
          -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
          -1.0, 2.0, 3.0, -4.0, 5.0, 6.0
        )
```

Reply:
Yeah I did the math by hand now and indeed you are right. Just for this part, I tried calculating it on the python terminal and due to my mistake with multiplying np arrays, I got this wrong, my bad.